app/controllers/admin/articles_controller.rb

```diff
@@ -44,6 +45,7 @@ def publish
 
     def unpublish
       @article.unpublished!
+      @article.touch(:unpublished_at)
```

Comment:
I can't see where the `Article` method `unpublished!` is defined, but that looks like a write operation. Could that be updated to include an update of the `unpublished_at` field?

Reply:
Not sure myself where `unpublished?` is defined. There's a file `article_policy` in `app/policies` which I think might need to be modified, but again I'm not too sure. A lot of this is Brett's code so I was just working on some additions in this pr

Reply:
OK - leave it as is then.

Reply:
`enum` in rails defines a bunch of helper methods (i.e. `unpublished!` `unpublished?` and scope methods like `Article.unpublished`. I thought this was common knowledge, but Guy was also unaware - perhaps should avoid using in the future to avoid these type of surprises?  I just find them really convenient.

And yes, you can override it to combine the database writes, but I was going for clarity here (and following guy's lead of putting everything in the controller rather than the model) - happy for it to move to the model and for the method to be overwritten.

Reply:
Yes - I see it now. Thanks @ebrett 

I think that setting `unpublished_at` is so tightly coupled to `unpublished!` that the two should be tied together in the model. Perhaps like this in the model:

```ruby
def unpublished!
 self.unpublished_at = Time.now.utc
 super
end
```